src/libraries/System.IO.FileSystem/tests/FileSystemTest.Browser.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public abstract partial class FileSystemTest
+    {
+        protected static int geteuid()
+        {
+            throw new PlatformNotSupportedException();
```

Comment:
What do you think about returning 0 here and eliminating the platform checks?

Reply:
By doing that and modifying the places that use `geteuid` the tests fail with:

```
[FAIL] System.IO.Tests.File_AppendAllLinesAsync.WriteToReadOnlyFile
        
  [40m[37mdbug[39m[22m[49m: test[0]
        System.UnauthorizedAccessException : Access to the path '/tmp/File_AppendAllLinesAsync_mqajhnoo.uqp/WriteToReadOnlyFile_105_93d848c1' is denied.
        
  [40m[37mdbug[39m[22m[49m: test[0]
        ---- System.IO.IOException : Permission denied
```

Reply:
The pal uses `sendFile` and emscripten's implementation doesn't seem too friendly w/ read only files.  The error message refers to not being able to write to the destination file.

If you skip this block https://github.com/dotnet/runtime/blob/05f17e3170d014efe63202c4e7e7b641e8f8fef6/src/libraries/Native/Unix/System.Native/pal_io.c#L1152-L1191, I think copying will work.

Reply:
I have gone through the build and it looks like this code block is already being skipped in the native pal.

Reply:
You're right.  I was looking at the wrong place thinking it was included.  

We should trace to see what call is bad.

Reply:
agreed